client/src/components/SingleListing.tsx

```diff
@@ -0,0 +1,456 @@
+    let listingIdValue = stringFilter(listingId);
+    const [listingData, setListingData] = useState(undefined);
+    const [bookingsForListingData, setBookingsForListingData] = useState([]);
+    const [reviews, setReviews] = useState([]);
+    const [textValue, setTextValue] = useState('');
+    const [value, setValue] = useState<DateRange<Dayjs>>([null, null]);
+    const [reviewRating, setReviewRating] = useState<number | null>(1);
+    const [numberOfPeople, setNumberOfPeople] = useState(1);
+    const { currentUser } = useContext(AuthContext);
+    const navigate = useNavigate();
+    const [open, setOpen] = React.useState(false);
+    const [bookingId, setBookingId] = useState('');
+    const [reviewError, setReviewError] = useState(undefined);
+    const [bookingError, setBookingError] = useState(undefined);
+    const [listingError, setListingError] = useState([404, 'Error Not Found']);
+
+    const increasePeople = () => {
+        setNumberOfPeople(numberOfPeople + 1);
+    };
+
+    const decreasePeople = () => {
+        if (numberOfPeople > 1) setNumberOfPeople(numberOfPeople - 1);
+    };
+
+    const buildReviewCard = (review : {user : string, rating: number, text: string, date: string}) => {
+        return (
+            <Grid item sx={{pb: 4}} key={Math.random()}>
+                <Card>
+                    <CardHeader
+                        avatar={
+                            <Avatar src="/broken-image.jpg"/>
+                        }
+                        title={review.user}
+                        subheader={review.date}
+                    >
+                    </CardHeader>
+                    <CardContent>
+                        <Rating 
+                            name="read-only" 
+                            value={review.rating}
+                            sx={{'& .MuiRating-iconFilled' : {
+                                color : '#000000'
+                            }}}
+                            readOnly 
+                        />
+                        <Typography>{review.text}</Typography>
+                    </CardContent>
+                </Card>
+            </Grid>
+        );
+    };
+
+    const bookListing = async () => {
+
+        if (currentUser === null) {
+            navigate('/login');
+            return;
+        }
+
+        if (value[0] === null || value[1] === null) {
+            setBookingError('You must select a start and end date');
+            return;
+        }
+
+        try {
+            const { data } = await axios.post('http://localhost:4000/bookings/create', {
+                bookerId : currentUser.uid,
+                listingId : listingData.listingId,
+                numOfPeople : numberOfPeople,
+                endTimestamp : value[1].toDate(),
+                startTimestamp : value[0].toDate()
+            });
+
+            setBookingId(data.bookingId);
+            setOpen(true);
+        } catch (e) {
+            console.log(e);
```

Comment:
So when I tried to create a booking on a already booked date, the server did through the 409 error but the client didn't display it. I think this just need to needs to be changed to an alert.

Reply:
- added backend logic to increase numOfBookings in listings collection when a new booking is created.
- added alert when backend throws an error when trying to create a new booking (additionally removed console log of error) 
Also FYI, you can technically put in a date on the date range picker but if you dont click in on the popup, there is technically no value hence backend throwing invalid date since the endTimestamp is null. Values must be selected through picker. 
![image](https://user-images.githubusercontent.com/52805736/208140693-657018a3-543c-4f95-a940-0b1a6b60d125.png)